pilot/pkg/model/push_context.go

```diff
@@ -749,6 +755,16 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 	}
 }
 
+// Caches list of authentication policies
+func (ps *PushContext) initAuthPolicies(env *Environment, services []*Service) {
+	for _, svc := range services {
+		ps.AuthPolicies[svc.Hostname] = map[int]*Config{}
+		for _, port := range svc.Ports {
```

Comment:
If we want to support label-selector, we may need a map from label-set to policy. I guess we can delay this, given it's still up in the air for the authn policy.